Sources/SourceKitLSP/Swift/SwiftLanguageService.swift

```diff
@@ -741,9 +741,19 @@ extension SwiftLanguageService {
         """
     }
 
+    var tokenRange: Range<Position>?
+
+    if let symbolUnderCursor = cursorInfoResults.first,
+      let line = symbolUnderCursor.symbolInfo.line,
+      let column = symbolUnderCursor.symbolInfo.column,
+      let length = symbolUnderCursor.symbolInfo.length
+    {
+      tokenRange = Position(line: line, utf16index: column)..<Position(line: line, utf16index: column + length)
+    }
```

Comment:
Column and length returned by sourcekitd are UTF-8, while `Position` is UTF-16. There are methods on `DocumentSnapshot` to convert between the two representations.